src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -44,4 +45,22 @@ private String getErrorMessage(ObjectError e) {
         }
         return e.getDefaultMessage();
     }
+
+    @ExceptionHandler(RegistrationException.class)
+    protected ResponseEntity<Object> handleRegistrationException(Exception exception) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        body.put("errors", exception.getMessage());
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(Exception exception) {
+        Map<String, Object> body = new LinkedHashMap<>();

```

Comment:
A lot of code duplication. Create a separate private method which builds the map.

Reply:
Now you send `BAD_REQUEST` status when `EntityNotFoundException` was thrown. But you should send `NOT_FOUND` status for this case. Also you can specify `@ExceptionHandler` by array of exceptions.